### PR TITLE
Purge blockstore evm data

### PR DESCRIPTION
**Pull request by gBaGu**
Implement purging of evm blockstore data

Add Block header index by slot to be able to purge evm headers using slot range.

### CODE REVIEW

File: ledger/src/blockstore_db.rs [hunk 1]
```diff
@@ -125,55 +126,55 @@ pub enum IteratorMode<Index> {
 pub mod columns {
     #[derive(Debug)]
     /// The slot metadata column
-    pub struct SlotMeta;
+    pub struct SlotMeta;//
```
Comment (vldm) [inline ledger/src/blockstore_db.rs]: what this mark for?

File: ledger/src/blockstore_db.rs [hunk 2]
```diff
@@ -456,7 +468,9 @@ impl Rocks {
             for cf_name in cf_names {
                 // this special column family must be excluded from LedgerCleanupService's rocksdb
                 // compactions
-                if cf_name == TransactionStatusIndex::NAME {
+                if cf_name == TransactionStatusIndex::NAME || cf_name == EvmTransactionReceipts::NAME
```
Comment (vldm) [inline ledger/src/blockstore_db.rs:487]: probably this patch can be removed (it was implemented to avoid purging in this columns)

Comment (vldm) [reply ledger/src/blockstore_db.rs:487]: maybe `PurgedSlotFilter` should be rewritten to cleanup for block numbers

File: ledger/src/blockstore/blockstore_purge.rs [hunk 3]
```diff
@@ -129,73 +129,107 @@ impl Blockstore {
...
                 .db
-                .delete_range_cf::<cf::BlockHeight>(&mut write_batch, from_slot, to_slot)
+                .delete_range_cf::<cf::BlockHeight>(&mut write_batch, from_slot, to_slot_inclusive)
+                .is_ok()
+            & self
+                .db
+                .delete_range_cf::<cf::EvmHeaderIndexBySlot>(&mut write_batch, from_slot, to_slot_inclusive)
+                .is_ok();
+
+        let first_slot = self
```
Comment (vldm) [inline ledger/src/blockstore/blockstore_purge.rs]: as I understand it should return block number rather then slot?

Please add some test for cases where slots has offset +100 over block numbers

File: ledger/src/blockstore_db.rs (hunk 1 above)
Comment (gBaGu) [reply ledger/src/blockstore_db.rs]: Removed

File: ledger/src/blockstore_db.rs (hunk 2 above)
Comment (gBaGu) [reply ledger/src/blockstore_db.rs:487]: Implemented compaction filter for evm blocks

File: ledger/src/blockstore/blockstore_purge.rs (hunk 3 above)
Comment (gBaGu) [reply ledger/src/blockstore/blockstore_purge.rs]: Refactored this part of code. Added more tests

**Pull request merged**